backend/task/update_account.go

```diff
@@ -39,27 +41,80 @@ func (task UpdateAccount) DoTask(fn func(string) chan bool) error {
 		return err
 	}
 
-	for i, val := range accounts {
-		if !strings.HasPrefix(val.Address, conf.Get().Hub.Prefix.AccAddr) {
-			continue
+	for i := range accounts {
+		if err := updateAccount(&accounts[i]); err != nil {
+			logger.Warn("get AccountInfo failed", logger.String("err", err.Error()))
 		}
-		_, _, rewards, err := lcd.GetDistributionRewardsByValidatorAcc(val.Address)
-		if err == nil && len(rewards) > 0 {
-			newrewards := loadRewards(rewards)
-			subvalue := newrewards.Amount - accounts[i].Rewards.Amount
+	}
 
-			accounts[i].Total.Amount += subvalue
+	return nil
+}
 
-			accounts[i].Rewards = newrewards
-		}
-		if err := task.account.Update(accounts[i]); err != nil {
-			logger.Warn("Account Update Rewards failed", logger.String("err", err.Error()))
-		}
+func updateAccount(account *document.Account) error {
+	if res, err := updateAccountInfo(account); err == nil {
+		account = res
+	} else {
+		logger.Warn("get AccountInfo failed", logger.String("err", err.Error()))
+	}
+	_, _, rewards, err := lcd.GetDistributionRewardsByValidatorAcc(account.Address)
+	if err == nil && len(rewards) > 0 {
+		newrewards := loadRewards(rewards)
+		subvalue := newrewards.Amount - account.Rewards.Amount
+		account.Total.Amount += subvalue
+		account.Rewards = newrewards
+	}
+	if err := account.Update(*account); err != nil {
+		logger.Warn("Account Update Rewards failed", logger.String("err", err.Error()))
+		return err
 	}
-
 	return nil
 }
 
+func updateAccountInfo(account *document.Account) (*document.Account, error) {
+	balance, res, err := getBalance(account)
+	if err != nil {
+		logger.Error("lcd getBalance Info have error", logger.String("err", err.Error()))
+		return account, err
+	} else {
+		account = res
+	}
+
+	delegate, err := getDelegationInfo(account.Address)
+	if err != nil {
+		logger.Warn("update Delegation Info have error", logger.String("err", err.Error()))
+	}
+	unbondingdelegator, err := getUnbondingDelegationInfo(account.Address)
+	if err != nil {
+		logger.Warn("update UnbondingDelegation Info have error", logger.String("err", err.Error()))
+	}
+	starttime := time.Unix(account.TotalUpdateAt, 0).In(cstZone)
```

Comment:
this is no need to update `create_at` and `height`, consider whether can remove these fields